src/main/java/de/tum/in/www1/artemis/repository/CompetencyRepository.java

```diff
@@ -43,29 +41,14 @@ public interface CompetencyRepository extends JpaRepository<Competency, Long> {
             """)
     Optional<Competency> findByIdWithLectureUnits(@Param("competencyId") long competencyId);
 
-    /**
-     * Fetches a competency with all linked exercises, lecture units, the associated progress, and completion of the specified user.
-     * <p>
-     * IMPORTANT: We use the entity graph to fetch the lazy loaded data. The fetched data is limited by joining on the user id.
-     *
-     * @param competencyId the id of the competency that should be fetched
-     * @param userId       the id of the user whose progress should be fetched
-     * @return the competency
-     */
     @Query("""
-            SELECT competency
-            FROM Competency competency
-                LEFT JOIN competency.userProgress progress
-                    ON competency.id = progress.competency.id AND progress.user.id = :userId
-                LEFT JOIN FETCH competency.exercises
-                LEFT JOIN FETCH competency.lectureUnits lectureUnits
-                LEFT JOIN lectureUnits.completedUsers completedUsers
-                    ON lectureUnits.id = completedUsers.lectureUnit.id AND completedUsers.user.id = :userId
-                LEFT JOIN FETCH lectureUnits.lecture
-            WHERE competency.id = :competencyId
+            SELECT c
+            FROM Competency c
+                LEFT JOIN FETCH c.lectureUnits lu
+                LEFT JOIN FETCH c.exercises
+            WHERE c.id = :competencyId
             """)
-    @EntityGraph(type = LOAD, attributePaths = { "userProgress", "lectureUnits.completedUsers" })
-    Optional<Competency> findByIdWithExercisesAndLectureUnitsAndProgressForUser(@Param("competencyId") long competencyId, @Param("userId") long userId);
+    Optional<Competency> findWithLectureUnitsAndExercisesById(@Param("competencyId") long competencyId);
```

Comment:
why can you not join more data directly in the query any more?

Reply:
I can join the data, however, then I would retrieve the data of all students and would have to remove all data not relating to the current student. Besides that it's technically a less secure approach, it was my original approach. @MaximilianAnzinger and @Strohgelaender correctly advised me that this would mean to include potentially thousands of progresses depending on the course size and asked me to retrieve them at least directly.